src/data/finsupp/basic.lean

```diff
@@ -1819,8 +1819,63 @@ begin
   exact h b (hb.2.symm ▸ ha)
 end
 
-lemma map_domain_comap_domain [add_comm_monoid M] (f : α → β) (l : β →₀ M)
-  (hf : function.injective f) (hl : ↑l.support ⊆ set.range f):
+section f_injective
+
+section has_zero
+variables [has_zero M]
+
+/-- Note the `hif` argument is needed for this to work in `rw`. -/
+@[simp] lemma comap_domain_zero (f : α → β)
+  (hif : set.inj_on f (f ⁻¹' ↑((0 : β →₀ M).support)) := set.inj_on_empty _) :
+  comap_domain f (0 : β →₀ M) hif = (0 : α →₀ M) :=
+by { ext, refl }
+
+@[simp] lemma comap_domain_single (f : α → β) (a : α) (m : M)
+  (hif : set.inj_on f (f ⁻¹' (single (f a) m).support)) :
+  comap_domain f (finsupp.single (f a) m) hif = finsupp.single a m :=
+begin
+  obtain rfl | hm := eq_or_ne m 0,
+  { simp_rw [single_zero],
+    rw comap_domain_zero },
+  { simp_rw [support_single_ne_zero hm, coe_singleton, set.preimage, set.mem_singleton_iff] at hif,
+    ext x,
+    obtain rfl | hx := eq_or_ne a x,
+    { rw [comap_domain_apply, single_eq_same, single_eq_same] },
+    { rw [comap_domain_apply, single_eq_of_ne (mt _ hx), single_eq_of_ne hx],
+      intro ax,
+      refine hif rfl ax.symm ax } },
```

Comment:
If you add `@[simps support]` to `def comap_domain`, then this proof can be golfed to:
```suggestion
  { simp only [single_zero, comap_domain_zero] },
  { rw [eq_single_iff, comap_domain_apply, comap_domain_support, ← finset.coe_subset, coe_preimage,
      support_single_ne_zero hm, coe_singleton, coe_singleton, single_eq_same],
    rw [support_single_ne_zero hm, coe_singleton] at hif,
    exact ⟨λ x hx, hif hx rfl hx, rfl⟩ }
```
Feel free to discard this suggestion if you don't like it.

Reply:
Yuri, thanks a lot!  I implemented your suggestion!  I will wait for CI to finish and merge!